app/models/collection.rb

```diff
@@ -1,5 +1,6 @@
 class Collection < ApplicationRecord
-  has_many :articles
+  has_many :articles, dependent: :nullify
```

Comment:
And for this one, it just sets associated datas foreign keys to null, correct? https://guides.rubyonrails.org/association_basics.html

Reply:
Yep! It means that if we call `collection.destroy` the `collection_id` values in `collection.articles` will be set to null